### PR TITLE
Use preact render instead of render-to-string from react

### PR DESCRIPTION
**Pull request by gtrufitt**
I was struggling with an error in Dotcom Rendering when trying to upgrade to Webpack 5:

```
  ERROR in ./node_modules/@guardian/atoms-rendering/dist/index.modern.js 1:771-821
  Module not found: Error: Package path ./compat/server is not exported from package /Users/gareth_trufitt/code/dotcom-rendering/node_modules/preact (see exports field in /Users/gareth_trufitt/code/dotcom-rendering/node_modules/preact/package.json)
```

The error came from the use of `import { renderToString } from 'react-dom/server';` in atoms-rendering, but I couldn't for the life of me get Dotcom Rendering to compile it properly, despite the use of it in DCR. I upgraded babel, webpack, preact, preact-render-to-string in both projects, messed around with webpack alias, ensured we were bundling etc but it just wouldn't work.

While this should be functionally identical, the problem will come if anyone uses atoms-rendering in a React project then they'll have to install preact *but* react-dom is [much (much) bigger](https://bundlephobia.com/result?p=react-dom@17.0.1) than [preact](https://bundlephobia.com/result?p=preact@10.5.7) and [preact-render-to-string](https://bundlephobia.com/result?p=preact-render-to-string@5.1.12) - so I am not particularly concerned...

### CODE REVIEW

**Comment by rcrphillips** [comment]
[Upgrade to Webpack 5](https://trello.com/c/8aUYN7dw/2122-upgrade-to-webpack-5)

**Review by webb04 — APPROVED** [review]
We're not using `preact` in apps-rendering but there's good reasons to switch over to it so happy for this to be merged.

We had a few problems upgrading to Webpack 5 too, I think our tripping point was with Storybook so would be good to see the dcr pull request!

 https://github.com/guardian/apps-rendering/pull/927

**Comment by gtrufitt** [comment]
> We're not using `preact` in apps-rendering but there's good reasons to switch over to it so happy for this to be merged.

Yeah, I think it only adds like 6kb to your build *if you don't have preact already* so it's not hugely painful

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
         "type": "ssh",
         "url": "https://github.com/guardian/atoms-rendering"
     },
-    "version": "2.4.1",
+    "version": "3.0.0-0",
```
Comment (oliverlloyd) [inline package.json:7]: Was this change intentional?

**Review by oliverlloyd — APPROVED** [review]
Happy to lean into preact more.

One thing though, do we need to be careful of this: https://github.com/preactjs/preact-render-to-string/issues/18?

**Comment by gtrufitt** [comment]
> One thing though, do we need to be careful of this: [preactjs/preact-render-to-string#18](https://github.com/preactjs/preact-render-to-string/issues/18)?

I don't think so? Html attributes are [case insensitive](https://w3c.github.io/html-reference/documents.html#case-insensitivity) so I think this is probably more of style thing?

File: package.json (hunk 1 above)
Comment (gtrufitt) [reply package.json:7]: Nope, I was trying to do a secret alpha release and forgot there was a prerelease script thing.

**Review by liywjl — APPROVED** [review]
Good workaround 👍  super strange edge case thought.
